buildkite/src/Jobs/Unit/Pipeline.dhall

```diff
@@ -0,0 +1,41 @@
+let Prelude = ../../../External/Prelude.dhall
+
+let Decorate = ../../../Lib/Decorate.dhall
+
+let Pipeline = ../../../Pipeline/Dsl.dhall
+let Command = ../../../Command/Base.dhall
+let Docker = ../../../Command/Docker/Type.dhall
+let Size = ../../../Command/Size.dhall
+let JobSpec = ../../../Pipeline/JobSpec.dhall
+
+let commands =
+  [
+    "./scripts/skip_if_only_frontend_or_rfcs.sh bash -c 'source ~/.profile && (dune runtest src/nonconsensus --profile=nonconsensus_medium_curves -j8 || (./scripts/link-coredumps.sh && false))'",
+    "./scripts/skip_if_only_frontend_or_rfcs.sh bash -c 'source ~/.profile && export GO=/usr/lib/go/bin/go && make build && (dune runtest src/lib --profile=dev_medium_curves -j8 || (./scripts/link-coredumps.sh && false))'",
+    "./scripts/skip_if_only_frontend_or_rfcs.sh bash -c 'source ~/.profile && make build && (dune runtest src/lib --profile=dev -j8 || (./scripts/link-coredumps.sh && false))'"
+  ]
+
+in
+
+Pipeline.build
+  Pipeline.Config::{
+    spec = ./Spec.dhall,
+    steps = [
+    Command.build
+      Command.Config::{
+        commands = Decorate.decorateAll commands,
+        label = "Unit Tests: dev, dev_medium_curves, nonconsensus_medium_curves",
+        key = "unit",
+        target = Size.Large,
+        docker = Docker::{
+          image = (../../../Constants/ContainerImages.dhall).toolchainBase,
+          environment = [
+            "BUILDKITE_AGENT_ACCESS_TOKEN",
```

Comment:
Since we always need `BUILDKITE_AGENT_ACCESS_TOKEN`, we should inject that via a Component.

This can be achieved in two different ways:
1. Have `Command.build` prepend `[ "BUILDKITE_AGENT_ACCESS_TOKEN" ] # docker.environment`
2. Add a `Docker/Base.dhall` that has a `Docker.build` that sticks the env in for you

(2) is more complex, so maybe we avoid doing that until we need more automation around the Docker setup.

Reply:
Copy that and pretty much went with *1.* while adding a `Constants` type for *ContainerEnvVars* (much like *ContainerImages*).